File: 코딩 기초 트레이닝/문자열 여러번 뒤집기-p181913/문자열 여러번 뒤집기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181913

def solution(my_string, queries):
    for query in queries:
        s = query[0]
        e = query[1]

        a = my_string[:s]
        b = my_string[s:e+1][::-1]
        c = my_string[e+1:]

        my_string = a + b + c

    return my_string

File: 코딩 기초 트레이닝/문자열 여러번 뒤집기-p181913/문자열 여러번 뒤집기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181913

function solution(my_string, queries) {

    queries.forEach((q) => {
        let s = q[0];
        let e = q[1];

        let a = my_string.slice(0, s);
        let b = my_string.slice(s, e+1).split("").reverse().join('');
        let c = my_string.slice(e+1,);

        my_string = a + b + c;
    });

    return my_string;
}
